sdks/java/io/cdap/src/main/java/org/apache/beam/sdk/io/cdap/CdapIO.java

```diff
@@ -17,29 +17,165 @@
+ *
+ * <p>To create the object of the {@link Plugin} class with the {@link Plugin#createBatch(Class,
+ * Class, Class)} method, need to specify the following parameters:
+ *
+ * <ul>
+ *   <li>{@link io.cdap.cdap.etl.api.batch.BatchSink} class
+ *   <li>{@link OutputFormat} class
+ *   <li>{@link io.cdap.cdap.api.data.batch.OutputFormatProvider} class
+ * </ul>
+ *
+ * <p>For more information about the OutputFormat and OutputFormatProvider, see {@link
+ * HadoopFormatIO}.
+ *
+ * <p>Example of {@link CdapIO#write()} usage:
+ *
+ * <pre>{@code
+ * Pipeline p = ...; // Create pipeline.
+ *
+ * // Get or create data to write
+ * PCollection<KV<String, String>> input = p.apply(Create.of(data));
+ *
+ * // Create PluginConfig for specific plugin
+ * EmployeeConfig pluginConfig =
+ *         new ConfigWrapper<>(EmployeeConfig.class).withParams(TEST_EMPLOYEE_PARAMS_MAP).build();
+ *
+ * // Write using CDAP batch plugin
+ * input.apply(
+ *         "WriteBatch",
+ *         CdapIO.<String, String>write()
+ *             .withCdapPlugin(
+ *                 Plugin.createBatch(
+ *                     EmployeeBatchSink.class,
+ *                     EmployeeOutputFormat.class,
+ *                     EmployeeOutputFormatProvider.class))
+ *             .withPluginConfig(pluginConfig)
+ *             .withKeyClass(String.class)
+ *             .withValueClass(String.class)
+ *             .withLocksDirPath(tmpFolder.getRoot().getAbsolutePath()));
+ *     p.run();
+ * }</pre>
+ *
+ * <h2>Read from Cdap Plugin Streaming Source</h2>
+ *
+ * <p>To configure {@link CdapIO} source, you must specify Cdap {@link Plugin}, Cdap {@link
+ * PluginConfig}, key and value classes.
+ *
+ * <p>{@link Plugin} is the Wrapper class for the Cdap Plugin. It contains main information about
+ * the Plugin. The object of the {@link Plugin} class can be created with the {@link
+ * Plugin#createStreaming(Class)} method. Method requires {@link
+ * io.cdap.cdap.etl.api.streaming.StreamingSource} class parameter.
+ *
+ * <p>Every Cdap Plugin has its {@link PluginConfig} class with necessary fields to configure the
+ * Plugin. You can set the {@link Map} of your parameters with the {@link
+ * ConfigWrapper#withParams(Map)} method where the key is the field name.
+ *
+ * <p>For example, to create a basic {@link CdapIO#read()} transform:
+ *
+ * <pre>{@code
+ * Pipeline p = ...; // Create pipeline.
+ *
+ * // Create PluginConfig for specific plugin
+ * EmployeeConfig pluginConfig =
+ *         new ConfigWrapper<>(EmployeeConfig.class).withParams(TEST_EMPLOYEE_PARAMS_MAP).build();
+ *
+ * // Read using CDAP streaming plugin
+ * p.apply("ReadStreaming",
+ * CdapIO.<String, String>read()
+ *             .withCdapPlugin(Plugin.createStreaming(EmployeeStreamingSource.class))
+ *             .withPluginConfig(pluginConfig)
+ *             .withKeyClass(String.class)
+ *             .withValueClass(String.class));
+ * }</pre>
  */
 @Experimental(Kind.SOURCE_SINK)
+@SuppressWarnings("nullness")
```

Comment:
Please fix null errors instead of supressing.

Reply:
You can reproduce checkerframework errors locally by temporarily setting following to false.
https://github.com/apache/beam/blob/b7b71361590b4fa8bac32a4541058bafdf0d1df1/buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy#L959

Reply:
Fixed